src/TLuaInterpreter.cpp

```diff
@@ -15449,6 +15450,71 @@ int TLuaInterpreter::getConnectionInfo(lua_State *L)
     return 2;
 }
 
+int TLuaInterpreter::unzip2(lua_State *L)
+{
+    QString zipLocation;
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "unzip2: bad argument #1 type (zip location as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    } else {
+        zipLocation = QString::fromUtf8(lua_tostring(L, 1));
+    }
+
+    QString extractLocation;
+    if (!lua_isstring(L, 2)) {
+        lua_pushfstring(L, "unzip2: bad argument #2 type (extract location as string expected, got %s!)", luaL_typename(L, 2));
+        return lua_error(L);
+    } else {
+        extractLocation = QString::fromUtf8(lua_tostring(L, 2));
+    }
+
+    QTemporaryDir temporaryDir;
+    if (!temporaryDir.isValid()) {
+        lua_pushnil(L);
+        lua_pushstring(L,
+                       "coudln't create temporary directory to extract the zip into");
+        return 2;
+    }
+
+    extractLocation = QDir::fromNativeSeparators(extractLocation);
+    if (!extractLocation.endsWith(QLatin1String("/"))) {
+        extractLocation.append(QLatin1String("/"));
+    }
+
+    QDir dir;
+    if (!dir.mkpath(extractLocation)) {
+        lua_pushnil(L);
+        lua_pushstring(L,
+                       "coudln't create output directory to extract the zip into");
+        return 2;
+    }
+
+    auto future = QtConcurrent::run(mudlet::unzip, zipLocation, extractLocation, temporaryDir.path());
+    auto watcher = new QFutureWatcher<bool>;
+    QObject::connect(watcher, &QFutureWatcher<bool>::finished, [=]() {
+        TEvent event {};
+        Host& host = getHostFromLua(L);
+
+        if (future.result()) {
+            event.mArgumentList.append(QStringLiteral("sysUnzipDone"));
+            event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+        } else {
+            event.mArgumentList.append(QStringLiteral("sysUnzipError"));
```

Comment:
In this case is it possible to capture and return any `libzip` error message (I know that we are still a bit deficient in other places where we are working with archive files in returning error messages but let's do it better here...)?

Reply:
The `mudlet::unzip()` function doesn't return it, but that can be added at a later point...

Reply:
:disappointed: Okay, guess that will have to be deferred...